### PR TITLE
[local-authentication] Add isAvailableAsync and tests

### LINKED ISSUE #9653: [expo-local-authentication] Create isAvailableAsync method
It's important that each module has a method for getting native availability.

### PR DESCRIPTION
**Pull request by akinncar**
# Why

Fixes #9653 

# How

Added isAvailableAsync() method, tests, and update docs.

# Test Plan

Added unit tests.

### CODE REVIEW

**Review by lukmccall — APPROVED** [review]
LGTM 👍 Greate job 💪

File: docs/pages/versions/unversioned/sdk/local-authentication.md [hunk 1]
```diff
@@ -29,6 +29,14 @@ import * as LocalAuthentication from 'expo-local-authentication';
 
 ## Methods
 
+### `LocalAuthentication.isAvailableAsync()`
+
+Returns whether the Local Authentication API is enabled on the current device.
```
Comment (tsapeta) [inline docs/pages/versions/unversioned/sdk/local-authentication.md:34]:
```suggestion
Returns whether the native Local Authentication API is available on the current platform.
```

File: docs/pages/versions/unversioned/sdk/local-authentication.md [hunk 2]
```diff
@@ -29,6 +29,14 @@ import * as LocalAuthentication from 'expo-local-authentication';
...
+
+#### Returns
+
+Async `boolean`, indicating whether the Local Authentication API is avaliable on the current device. Currently this resolves `true` on iOS and Android only.
```
Comment (tsapeta) [inline docs/pages/versions/unversioned/sdk/local-authentication.md:38]:
```suggestion
Async `boolean`, indicating whether the Local Authentication API is available on the current platform. Currently this resolves to `true` on iOS and Android only.
```

File: packages/expo-local-authentication/src/LocalAuthentication.ts [hunk 3]
```diff
@@ -10,6 +10,13 @@ import {
 
 export { LocalAuthenticationOptions, AuthenticationType, LocalAuthenticationResult };
 
+/**
+ * Returns whether the Local Authentication API is enabled on the current device.
```
Comment (tsapeta) [inline packages/expo-local-authentication/src/LocalAuthentication.ts:14]:
```suggestion
 * Returns whether the native Local Authentication API is available on the current platform.
```

File: packages/expo-local-authentication/src/__tests__/LocalAuthentication-test.web.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import * as LocalAuthentication from '../LocalAuthentication';
+
+describe(`isAvailableAsync`, () => {
+  it('resolves to true on web platform', async () => {
+    await expect(LocalAuthentication.isAvailableAsync()).resolves.toBeTruthy();
+  });
```
Comment (tsapeta) [inline packages/expo-local-authentication/src/__tests__/LocalAuthentication-test.web.ts:6]: In fact it always returns `false` on Web 😄

**Review by tsapeta — COMMENTED** [review]
I'm not entirely sure what's the purpose of this PR and what's the distinction between `isAvailableAsync` and `hasHardwareAsync`.

I guess `isAvailableAsync` is intended to just let us know whether the API <ins>is **available** on the current **platform**</ins> instead of <ins>is **enabled** on the current **device**</ins>. The latter one is actually what `hasHardwareAsync` means and so we should emphasize the difference between them. With that in mind, I added some suggestions 👇

**Comment by brentvatne** [comment]
closing this because inactive :)

**Pull request closed without merging**